launchpad/deposit.gno

```diff
@@ -0,0 +1,578 @@
+    projects[deposit.projectId] = *project
+
+    return deposit.amount, nil
+}
+
+func CollectDepositGnsByDepositId(depositId string) uint64 {
+	common.IsHalted()
+    caller := std.PrevRealm().Addr()
+
+    // Validate deposit collection request
+    deposit, project, err := validateDepositCollection(depositId, caller)
+    if err != nil {
+        return 0
+    }
+
+    en.MintAndDistributeGns()
+
+    calculateDepositReward()
+    project = projects[deposit.projectId] // get updated project
+
+    // Process deposit collection
+    height := uint64(std.GetHeight())
+    amount, err := processDeposit(&deposit, &project, height)
+    if err != nil {
+        return 0
+    }
+
+    // Update gov_staker contract
+    gs.SetAmountByProjectWallet(project.recipient, amount, false)
+
+    // Emit collection event
+    prevAddr, prevRealm := getPrev()
+    std.Emit(
+        "CollectDepositGnsByDepositId",
+        "prevAddr", prevAddr,
+        "prevRealm", prevRealm,
+        "depositId", depositId,
+        "amount", ufmt.Sprintf("%d", amount),
+    )
+
+    // Process token transfers if amount > 0
+    if amount > 0 {
+        xgns.Burn(a2u(consts.LAUNCHPAD_ADDR), amount)
+        gns.Transfer(a2u(caller), amount)
+        return amount
+    }
+
+    return 0
+}
+
+func getProjectIdFromTierId(tierId string) string {
+	// input: gno.land/r/gnoswap/gns:123:30
+	// output: gno.land/r/gnoswap/gns:123
+
+	res, err := common.Split(tierId, ":", 3)
+	if err != nil {
+		panic(addDetailToError(
+			errInvalidTier,
+			ufmt.Sprintf("invalid tierId: %s", tierId),
+		))
+	}
+
+	return ufmt.Sprintf("%s:%s", res[0], res[1])
+}
+
+func getProjectIdAndTierFromTierId(tierId string) (string, string) {
+	res, err := common.Split(tierId, ":", 3)
+	if err != nil {
+		panic(addDetailToError(
+			errInvalidTier,
+			ufmt.Sprintf("invalid tierId: %s", tierId),
+		))
+	}
+
+	return ufmt.Sprintf("%s:%s", res[0], res[1]), res[2]
+}
+
+func checkDepositConditions(project Project) {
+	if project.conditions == nil {
+		return
+	}
+
+	caller := std.PrevRealm().Addr()
+	for _, condition := range project.conditions {
+		if condition.minAmount == 0 {
+			continue
+		}
+		
+		// check balance
+		var balance uint64
+		if condition.tokenPath == consts.GOV_XGNS_PATH {
+			balance = xgns.BalanceOf(a2u(caller))
+		} else {
+			balance = common.BalanceOf(condition.tokenPath, caller)
+		}
+		if balance < condition.minAmount {
+			panic(addDetailToError(
+				errNotEnoughBalance,
+				ufmt.Sprintf("insufficient balance(%d) for token(%s)", balance, condition.tokenPath),
+			))
+		}
+	}
+}
+
+// Deprecated: use isProjectActive instead
+func checkProjectActive(project Project) bool {
+	if project.started.height > uint64(std.GetHeight()) {
+		// not started yet
+		return false
+	}
+
+	if project.ended.height < uint64(std.GetHeight()) {
+		// already ended
+		return false
+	}
+
+	return true
+}
+
+// Deprecated: use isTierActive instead
+func checkTierActive(project Project, tier Tier) bool {
```

Comment:
let's deprecated function.

Reply:
It's not actually a depreacatede function, so I just removed the comment.

https://github.com/gnoswap-labs/gnoswap/pull/451/commits/f5b33c63e8245d54020a32e7c2bf171cbdf77e86